### DISCUSSION TITLE
Is it expected that Gotenberg 8 uses 6x memory of Gotenberg 7

### DISCUSSION THREAD

**Discussion by tommoor** [gotenberg/gotenberg#1057] [Q&A]
I recently decided to upgrade from 7 as part of debugging some print overflow issues, I checked the compatibility document and everything looked good for our simple usage – but immediately after deploying the image I had to downgrade as memory usage shot through the roof and requests began timing out.

### Before

<img width="361" alt="image" src="https://github.com/user-attachments/assets/6f840638-a2d0-4a96-9b9b-c8fb5155e50b">

### After

<img width="393" alt="image" src="https://github.com/user-attachments/assets/7ac5bfdf-2fbb-4805-b79b-d907adb0431a">

**Comment by gulien** [comment]
Hey @tommoor 👋 

I don't know which exact `7.x` version of Gotenberg you've been using, but since [7.10.0](https://github.com/gotenberg/gotenberg/releases/tag/v7.10.0), there were a lot of changes regarding processes management.

The TL;DR: there are now a single Chromium / LibreOffice processes. 

What are the impacts?

* Each of this process handle one, and only one conversion at a time. Scalability is now in the hands of the container orchestrator. If you're encountering timeouts, I'd suggest to spin up one or more instances of Gotenberg.
* Regarding the memory usage, I guess, if you were in stateless mode, there was no process currently running. Now you have always a Chromium and/or LibreOffice instance running, which might explain the increase.

Hope this helps 😉

**Reply by tommoor** [reply]
I'm also not sure what version of 7 was running honestly, as it was just the `gotenberg:7` tagged docker image from when it was last deployed a while back. Having moved back to `7.10.2` specifically the memory usage is still 2x what it was previously (~140Mb idle), but not as high as `8` (~470Mb idle).

> If you're encountering timeouts, I'd suggest to spin up one or more instances of Gotenberg.

This was just with a single request to generate a very small PDF. I suspect that the box ran out of memory whereas before it handled this easily. I will probably stay on `7.x` for now as it allows using containers that are half the size.

**Reply by gulien** [reply]
> This was just with a single request to generate a very small PDF

That's because a Chromium process was spawn, nothing fancy; it won't take much more for subsequent requests. For instance, the demo runs on 500MB and handles 30k requests per week without hassle.
